accepted/2020/linking-libraries.md

```diff
@@ -1,170 +1,572 @@
+The developer can also enable trim analysis to gain confidence that trimming
+will not introduce unexpected changes to the application's behavior. The feedback
+presented by the analysis must be actionable in the context of the application.
 
 ### Goals
 
 1. Ensure all required code is preserved by the linker.
-2. Allow Xamarin and Blazor applications to use the .NET Libraries and be as small as possible.
-
-### Non-Goals
-
-1. Optimizing application size in other form factors.
+1. Application size is as small as possible.
+1. Trim analysis feedback is actionable for application developers.
 
 ## Design
 
-There are a number of work items the .NET Libraries Team needs to undertake in order to be linker-friendly. In general they fall into the two buckets outlined above:
-
-1. Linker-safety
-2. Smaller Size
-
-### Linker-safety
-
-Ensuring required code is preserved by the linker is not easy. .NET has a very rich Reflection capability that enables dynamic invocation of code. There are many cases where a static analysis tool (like the ILLinker) is not able to know which code is required and which isn't. For example, a .NET application could read a Type name from a file and create a new instance of that Type through Reflection. When the linker analyzes the application statically, it doesn't see any hard references to that Type and removes it. But when the application is executed, an error occurs because the Type is no longer available in the application.
-
-There are features in the .NET Libraries that will be very expensive, and in some cases virtually impossible, to make 100% linker-safe. A few examples are:
-- `System.ComponentModel.Composition`, `System.Composition`, and other dependency injection style libraries
-    - Contains features such as loading "parts" from a directory
-- `Microsoft.CSharp` and `System.Linq.Expressions`
-    - Use Reflection extensively to implement `dynamic` and expression trees
-- `System.Resources.ResourceManager`
-    - Resource files can have ResourceReader Type names embedded in them, which `ResourceManager` creates dynamically.
-- `Assembly.Load/LoadFrom/LoadFile`
-    - Dynamically loading assemblies into a trimmed application is problematic because if the linker isn't aware of those assemblies, it won't know to preserve the methods required by those assemblies. If the core application doesn't need `string.Split`, the linker will trim the `Split` method away. When an assembly that the linker didn't know about is loaded dynamically, it may need `string.Split`, and fail to load.
-- `System.Reflection.Emit`
-    - A method or property that is only called by dynamically generated code may be trimmed if the linker isn't aware of it.
+There are a number of things the author of a library needs to undertake in order
+to make it work well with trimmed applications. In general they fall into
+the two buckets outlined above:
+
+1. Trimming compatibility
+1. Smaller Size
```

Comment:
```suggestion
1. Ensuring code required for correctness is preserved by the linker
1. Ensuring as much of the library as is possible is trimmable if not directly used
```
or something like that.  As written, these aren't really things that can be "undertaken".